core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -774,10 +774,10 @@ class KafkaApis(val requestChannel: RequestChannel,
         // which were written in the new format prior to the version downgrade.
         val unconvertedRecords = FetchResponse.recordsOrFail(partitionData)
         val downConvertMagic =
-          logConfig.map(_.messageFormatVersion.recordVersion.value).flatMap { magic =>
-            if (magic > RecordBatch.MAGIC_VALUE_V0 && versionId <= 1 && !unconvertedRecords.hasCompatibleMagic(RecordBatch.MAGIC_VALUE_V0))
+          logConfig.map(_.recordVersion.value).flatMap { magic =>
```

Comment:
Hmm, I thought we were going to simplify this. Basically ignore the log config and just rely on the fetch version. The intuition is that _all_ new data will need to be down-converted anyway. Perhaps we want to wait until the IBP is at 3.0 for this logic to kick in?

Reply:
With IBP 3.0, this will always be 3.0, so it works as you said. Even though it's a bit roundabout. I could have the IBP check here, but it seems to spread the logic into more places. I could add a comment perhaps. Thoughts?

Reply:
I updated the comment to reflect this.